src/user-management/interfaces/user.interface.ts

```diff
@@ -0,0 +1,27 @@
+// TODO
+// - should User be defined as `ManagedUser | UnmanagedUser` ?
```

Comment:
thoughts on this? typescript can model this nicer than having a bunch of optional fields, but leaving the User interface like this is probably easier for typescript novices?

Reply:
I think we should use a union type here.